packages/flutter/lib/src/cupertino/loupe.dart

```diff
@@ -0,0 +1,266 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+import 'package:flutter/widgets.dart';
+
+/// A CupertinoLoupe, specifically for text editing.
+///
+/// Delegates styling to [CupertinoLoupe] with is positioned depending on
+/// [loupeSelectionOverlayInfoBearer].
+///
+/// Specifically, the [CupertinoTextEditingLoupe] follows the following rules:
+/// - Never position itself horizontally outside the screen width, with [_kHorizontalScreenEdgePadding] padding.
+/// - Hidden if a gesture is detected [_kHideIfBelowThreshold] units below the line, shown otherwise.
+/// - Follow the X of the gesture directly (with respect to rule 1).
+/// - Have some vertical drag resistance; i.e. if a gesture is detected k units below the field,
```

Comment:
Tried to make them complete sentences, but it's hard without being too repetitive. It read like:

```dart
/// - [CupertinoTextEditingLoupe] positions itself horizontally outside the screen width,
/// with _kHorizontalScreenEdgePadding padding.
/// - [CupertinoTextEditingLoupe] is hidden if a gesture is detected _kHideIfBelowThreshold units below the line that the loupe is on, shown otherwise.
```

all starting with the same word. I'll keep them as incomplete sentences, but switch to singlar present tense.